src/Features/Lsif/Generator/Generator.cs

```diff
@@ -251,6 +251,45 @@ SymbolKind.RangeVariable or
                     {
                         var definitionResultsId = symbolResultsTracker.GetResultIdForSymbol(declaredSymbol, Methods.TextDocumentDefinitionName, () => new DefinitionResult(idFactory));
                         lsifJsonWriter.Write(new Item(definitionResultsId.As<DefinitionResult, Vertex>(), lazyRangeVertex.Value.GetId(), documentVertex.GetId(), idFactory));
+
+                        // If this declared symbol also implements an interface member, we count this as a definition of the interface member as well.
+                        // Note in C# there are estoeric cases where a method can implement an interface member even though the containing type does not
+                        // implement the interface, for example in this case:
+                        //
+                        //     interface I { void M(); }
+                        //     class Base { public void M() { } }
+                        //     class Derived : Base, I { }
+                        //
+                        // We don't worry about supporting these cases here.
+                        var implementedMembers = declaredSymbol.ExplicitOrImplicitInterfaceImplementations();
+
+                        foreach (var implementedMember in implementedMembers)
+                            MarkImplementationOfSymbol(implementedMember);
+
+                        // If this overrides a method, we'll also mark it the same way. We want to chase to the base virtual method, skipping over intermediate
+                        // methods so that way all overrides of the same method point to the same virtual method
+                        if (declaredSymbol.IsOverride)
+                        {
+                            var overridenMember = declaredSymbol.GetOverriddenMember();
+
+                            while (overridenMember?.GetOverriddenMember() != null)
+                                overridenMember = overridenMember.GetOverriddenMember();
+
+                            if (overridenMember != null)
+                                MarkImplementationOfSymbol(overridenMember);
+                        }
+
+                        void MarkImplementationOfSymbol(ISymbol baseMember)
```

Comment:
The only way this will make sense is if you look at the picture in https://github.com/dotnet/roslyn/issues/59617.